### ISSUE TITLE
Connection closed by broker on connect

### ISSUE THREAD

**Issue by meilon** [ckrey/MQTT-Client-Framework#20]
Hi,

I'd like to use MQTT-Client-Framework in my Swift iOS project, but I can't even connect. I wrote a singleton class to handle all my MQTT business in the app, and currently in the init() I'd like to connect. This is the code I use currently:

    init() {
        _mqttSession = MQTTSession(clientId: _clientId, userName: nil, password: nil, keepAlive: 10, cleanSession: false)
        _mqttSession.delegate = sessD
        _mqttSession.connectToHost(kMQTTServerHost, port: 1883, usingSSL: false)
    }

sessD is a class with the NSObject and MQTTSessionDelegate protocols. I've added all functions, but even though `connectionError` is implemented, it doesn't get called, only `handleEvent`.

I see two handleEvent calls, one `ConnectionClosed` and a `ConnectionClosedByBroker`. On the broker (mosquitto 1.3.5) I can see 

    1424447431: New connection from 78.43.156.253 on port 1883.
    1424447431: Socket error on client (null), disconnecting.

 So something seems to be wrong the way the client tries to connect.

So far I've added the xcodeproj as a sub project, added libMQTTClient.a to the "linked frameworks and libraries" list, created a <project-name>-Bridging-Header.h where I added `#import "MQTTClient/MQTTClient.h" and added it to the Build Settings Objective-C Bridging Header section.

What did I do wrong?

**Comment by ckrey** [comment]
It seems to me you are doing nothing wrong. But I can reproduce the problem.

I have not found the reason for the problem. It does not occur if you use the full initializer:
```
_mqttSession = MQTTSession(clientId: _clientId, userName: nil, password: nil, keepAlive: 10, cleanSession: false
        will: false,
        willTopic: nil,
        willMsg: nil,
        willQoS: MQTTQosLevel.QoSLevelAtMostOnce,
        willRetainFlag: false,
        protocolLevel: 4,
        runLoop: nil,
        forMode: nil
)
```

**Comment by ckrey** [comment]
the issue is not swift specific, happens in Objective-C too.
Will analyse and fix

**Comment by ckrey** [comment]
Problem found and fixed! Will be on github and cocoapods (0.0.5) in a second

*The short form of the initializer used the Qos level `AtLeastOnce` rather than `AtMostOnce`.*

This was detected by mosquitto checking MQTT 3.1.1 conformance:
```
If the Will Flag is set to 0 the Will QoS and Will Retain fields in the Connect Flags MUST be set to zero and the Will Topic and Will Message fields MUST NOT be present in the payload [MQTT-3.1.2-11].
```
Mosquitto correctly closed the connection because of the detected protocol breach.
